optuna/terminator/serror.py

```diff
@@ -0,0 +1,54 @@
+import abc
+
+import numpy as np
+
+from optuna.study.study import Study
+from optuna.trial._state import TrialState
+
+
+class BaseStatisticalErrorEvaluator(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def evaluate(
+        self,
+        study: Study,
+    ) -> float:
+        pass
+
+
+class CrossValidationStatisticalErrorEvaluator(BaseStatisticalErrorEvaluator):
+    def __init__(self, user_attr_key: str = "cv_scores") -> None:
+        self._user_attr_key = user_attr_key
+
+    def evaluate(
+        self,
+        study: Study,
+    ) -> float:
+        if len(study.get_trials(states=(TrialState.COMPLETE,))) == 0:
+            raise ValueError(
+                "Statistical error cannot be calculated because no trial has been completed."
+            )
+
+        cv_scores = study.best_trial.user_attrs[self._user_attr_key]
+
+        k = len(cv_scores)
+        scale = 1 / k + 1 / (k - 1)
+
+        var = scale * np.var(cv_scores)
+
+        return float(var)
```

Comment:
> As an alternative, we could avoid using user attributes completely. If we introduce some helper function/method that internally sets system attributes (instead of user attributes), we could do validation when setting.

I prefer this option. I changed the interface so that the user calls the `report_cross_validation_scores` function instead of setting the scores as a user attr. PTAL. https://github.com/optuna/optuna/pull/4405/commits/7ab54308033810383aff9b6b4cab138635b41bf5

Reply:
Also, the length of the list is now checked in `report_cross_validation_scores`. https://github.com/optuna/optuna/pull/4405/commits/d23413ce8ab831e534149acff16066cf32ad156d

Reply:
Thanks for the update, let me resolve this conversation.